Remotion/Validation/Core/Validators/ExactLengthValidator.cs

```diff
@@ -15,18 +15,78 @@
 // along with re-motion; if not, see http://www.gnu.org/licenses.
 // 
 using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
 using JetBrains.Annotations;
+using Remotion.FunctionalProgramming;
+using Remotion.Utilities;
 using Remotion.Validation.Implementation;
+using Remotion.Validation.Results;
 
 namespace Remotion.Validation.Validators
 {
-  public class ExactLengthValidator : LengthValidator
+  public class ExactLengthValidator : ILengthValidator
   {
+    public int Min { get; }
+
+    public int? Max { get; }
+
+    public string ErrorMessage { get; }
+
+    public ValidationMessage ValidationMessage { get; }
+
     public ExactLengthValidator (int length, [NotNull] ValidationMessage validationMessage)
-        : base(length, length, $"The value must have exactly {length} characters.", validationMessage)
+        : this(length, $"The value must have exactly {length} characters.", validationMessage)
     {
     }
 
-    // TODO RM-5906: build separate validator with a dedicated message
+    private ExactLengthValidator (int length, [NotNull] string errorMessage, [NotNull] ValidationMessage validationMessage)
+    {
+      ArgumentUtility.CheckNotNullOrEmpty(nameof(errorMessage), errorMessage);
+      ArgumentUtility.CheckNotNull(nameof(validationMessage), validationMessage);
+
+      if (length < 0)
+        throw new ArgumentOutOfRangeException("length", "Value cannot be less than zero.");
+
+      Min = length;
+      Max = length;
+      ErrorMessage = errorMessage;
+      ValidationMessage = validationMessage;
+    }
+
+    public IEnumerable<PropertyValidationFailure> Validate (PropertyValidatorContext context)
+    {
+      if (IsValid(context))
+        return Enumerable.Empty<PropertyValidationFailure>();
+
+      return EnumerableUtility.Singleton(CreateValidationError(context));
+    }
+
+    private bool IsValid (PropertyValidatorContext context)
+    {
+      var propertyValue = context.PropertyValue;
+
+      if (propertyValue is not string stringValue)
+        return true;
+
+      return stringValue.Length >= Min && (Max == null || !(stringValue.Length > Max));
```

Comment:
Length should be persisted in a dedicatd Length property (not nullable) and Min and Max implemented as explicit interface to better represent the semantics.